src/simplewallet/simplewallet.cpp

```diff
@@ -2242,6 +2242,28 @@ bool simple_wallet::set_ignore_fractional_outputs(const std::vector<std::string>
   return true;
 }
 
+bool simple_wallet::set_device_name(const std::vector<std::string> &args/* = std::vector<std::string>()*/)
+{
+  const auto pwd_container = get_and_verify_password();
+  if (pwd_container)
+  {
+    m_wallet->device_name(args[0]);
```

Comment:
Does not test whether there is actually a parameter.

Reply:
Testing now